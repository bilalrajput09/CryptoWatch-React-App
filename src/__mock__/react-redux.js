const useSelector = jest.fn();

export default { useSelector };
