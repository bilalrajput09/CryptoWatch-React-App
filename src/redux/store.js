import { configureStore } from '@reduxjs/toolkit';
import coinsSlice from './coinsSlice';

const store = configureStore({
  reducer: { coins: coinsSlice },
});

export default store;
