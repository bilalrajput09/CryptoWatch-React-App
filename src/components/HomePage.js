import { useDispatch, useSelector } from 'react-redux';
import InputSearch from './InputSearch';
import DisplayCoin from './DisplayCoin';
import styles from './styles/HomePage.module.css';
import { coinsActions } from '../redux/coinsSlice';

const HomePage = () => {
  const coinsArray = useSelector((state) => state.coins.coins);
  const dispatch = useDispatch();

  const filteredArray = useSelector((state) => state.coins.filteredArray);

  const inputSearchHandler = (e) => {
    e.preventDefault();
    const keyWord = e.target.value;
    if (!keyWord) {
      dispatch(coinsActions.resetFilteredArray(coinsArray));
    }
    if (keyWord !== '') {
      const coin = coinsArray.filter((c) => c.name.toLowerCase().startsWith(keyWord.toLowerCase()));
      dispatch(coinsActions.filterCoins(coin));
    }
  };

  return (
    <>
      <InputSearch inputSearchHandler={inputSearchHandler} />
      <main>
        <ul className={styles.coins_main_container}>
          {filteredArray.length !== 0
            ? filteredArray.map((coin) => (
              <DisplayCoin key={coin.id} coin={coin} />
            ))
            : coinsArray.map((coin) => (
              <DisplayCoin key={coin.id} coin={coin} />
            ))}
        </ul>
      </main>
    </>
  );
};

export default HomePage;
