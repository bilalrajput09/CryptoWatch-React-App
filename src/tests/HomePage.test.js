import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import HomePage from '../components/HomePage';
import store from '../redux/store';

test('Search input should render', () => {
  render(
    <Provider store={store}>
      <HomePage />
    </Provider>,
  );
  const searchInput = screen.getByPlaceholderText(/coin name/i);
  expect(searchInput).toBeInTheDocument();
});

test('render homepage correctly', () => {
  const componentTree = renderer
    .create(<Provider store={store} />)
    .toJSON();
  expect(componentTree).toMatchSnapshot();
});
