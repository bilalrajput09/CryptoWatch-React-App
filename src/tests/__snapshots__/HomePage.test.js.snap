// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render homepage correctly 1`] = `null`;
